packages/iov-bns/src/encode.ts

```diff
@@ -162,22 +162,23 @@ function buildSendTransaction(tx: SendTransaction): codecImpl.app.ITx {
 }
 
 function buildSwapOfferTx(tx: SwapOfferTransaction): codecImpl.app.ITx {
-  const hashed: SwapCounterTransaction = {
-    kind: "bcp/swap_counter",
-    creator: tx.creator,
-    recipient: tx.recipient,
-    amounts: tx.amounts,
-    timeout: tx.timeout,
-    hashCode: preimageIdentifier(tx.preimage),
+  return {
```

Comment:
This is a copy of `buildSwapCounterTx`

If we no longer have the preimage/hash distinction, I wonder why we need two types here...
They are both "offers" on the respective chains, the interpretation of them is only whether the client produced the preimage for the hash, or whether it copied the hash without knowing the preimage.

Can we simplify by removing types?

Reply:
What was importent for me was to not pass the preimage as part of a transaction, as it suggests the secret becomes public data at a point where it should be a client-side secret.

So yeah, we ended up with two transaction types that have the same fields and are implemented as the same thing on BNS. But will they always be the same transaction from the blockchain's point of view?

But even if we keep the two BCP transaction types, we can combine the implementation into

```ts
function buildSwapCounterTx(tx: SwapOfferTransaction | SwapCounterTransaction): codecImpl.app.ITx
```

Reply:
> So yeah, we ended up with two transaction types that have the same fields and are implemented as the same thing on BNS. But will they always be the same transaction from the blockchain's point of view?

The issue is that they are encoded the same on the blockchain, so when parsing you have to assume type A or B, regardless of what was written.

And since each chain is a word unto itself, what we have is basically:

`Offer -> Claim` OR `Offer -> Return`

on each side.

Reply:
There is no way to differentiate the two without client side knowledge of both chains and intent, which we cannot codify